source/azure_iot.c

```diff
@@ -0,0 +1,410 @@
+    {
+        return eAzureIoTOutOfMemory;
+    }
+
+    /* Setup index into the pcBase64name.  */
+    j = 0;
+
+    /* Compute the pcBase64name.  */
+    ulStep = 0;
+    i = 0;
+
+    while( j < ulLength )
+    {
+        /* Determine which step we are in.  */
+        if( ulStep == 0 )
+        {
+            /* Use first 6 bits of name character for index.  */
+            pcBase64name[ j++ ] = ( char ) _cAzureIoTBase64Array[ ( ( uint8_t ) pucName[ i ] ) >> 2 ];
+            ulStep++;
+        }
+        else if( ulStep == 1 )
+        {
+            /* Use last 2 bits of name character and first 4 bits of next name character for index.  */
+            pcBase64name[ j++ ] = ( char ) _cAzureIoTBase64Array[ ( ( ( ( uint8_t ) pucName[ i ] ) & 0x3 ) << 4 ) | ( ( ( uint8_t ) pucName[ i + 1 ] ) >> 4 ) ];
+            i++;
+            ulStep++;
+        }
+        else if( ulStep == 2 )
+        {
+            /* Use last 4 bits of name character and first 2 bits of next name character for index.  */
+            pcBase64name[ j++ ] = ( char ) _cAzureIoTBase64Array[ ( ( ( ( uint8_t ) pucName[ i ] ) & 0xF ) << 2 ) | ( ( ( uint8_t ) pucName[ i + 1 ] ) >> 6 ) ];
+            i++;
+            ulStep++;
+        }
+        else /* Step 3 */
+        {
+            /* Use last 6 bits of name character for index.  */
+            pcBase64name[ j++ ] = ( char ) _cAzureIoTBase64Array[ ( ( ( uint8_t ) pucName[ i ] ) & 0x3F ) ];
+            i++;
+            ulStep = 0;
+        }
+    }
+
+    /* Determine if the index needs to be advanced.  */
+    if( ulStep != 3 )
+    {
+        i++;
+    }
+
+    /* Now add the PAD characters.  */
+    while( ( ulPad-- ) && ( j < base64name_size ) )
+    {
+        /* Pad pcBase64name with '=' characters.  */
+        pcBase64name[ j++ ] = '=';
+    }
+
+    /* Put a NULL character in.  */
+    pcBase64name[ j ] = 0;
+
+    return eAzureIoTSuccess;
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTResult_t AzureIoT_Init()
+{
+    #if ( LIBRARY_LOG_LEVEL >= LOG_INFO )
+        az_log_set_message_callback( prvAzureIoTLogListener );
+    #endif
+
+    return eAzureIoTSuccess;
+}
+/*-----------------------------------------------------------*/
+
+void AzureIoT_Deinit()
+{
+    AZLogDebug( ( "AzureIoT_Deinit called \r\n" ) );
+}
+/*-----------------------------------------------------------*/
+
+AzureIoTResult_t AzureIoT_MessagePropertiesInit( AzureIoTMessageProperties_t * pxMessageProperties,
+                                                uint8_t * pucBuffer,
+                                                uint32_t ulAlreadyWrittenLength,
+                                                uint32_t ulBufferLength )
+{
+    az_span xPropertyBufferSpan = az_span_create( pucBuffer, ( int32_t ) ulBufferLength );
+    az_result xResult;
+
+    if( pxMessageProperties == NULL )
+    {
+        AZLogError( ( "AzureIoTMessagePropertiesInit failed: Invalid argument \r\n" ) );
```

Comment:
I remember John pointing out that we should be consistent with our logging to be ending with \r\n or not... in provisioning I removed all the \r\n. 
I am ok with either as long we are consistent with our story.

Reply:
removing is a good idea